### DISCUSSION TITLE
Search page content through GraphQL API

### DISCUSSION THREAD

**Discussion by marcuspuchalla** [requarks/wiki#7335] [Help / Questions]
I am looking for a way to search the content of my wiki pages through the api.
But I cant seem to find a doc describing how to do this. I can list all pages, like this.
```
query {
      pages {
          list (locale: "en")  {
              id,...
          }
      }
    }
```
But search endpoint only seems to work for searching users. And I can't select page content in the pages query either.
Can someone clarify if this is possible at all ?

**Comment by vowqg** [comment]
ln this way, you can search page content like the website.
```
query {
      pages {
           search(query: "te"){
              results{
                  id,
                  title,
                  path,
                  description,
                  locale
              }
            suggestions,
            totalHits
         }
      }
    }
```
But I'm not sure you can get the full document content with the returned id, because this id may not correspond to the document id. However, you can use the `singleByPath` API and the `path` field you just got to get the full documentation content. But the 'singleByPath' API requires administrator privileges.

```
query {
    pages { 
        singleByPath (path:"path/to/doc",locale: "en") { 
            id,
            content 
        }
    }
}
```
